### ISSUE TITLE
Advanced Medical System - Wake up an unconscious patient in field?

### ISSUE THREAD

**Issue by Knight1981** [acemod/ACE3#1329]
Hey guys,

I have a question / feature request. We would like to play with the advanced medical system, but with the advanced system it seems that we have no possibility to wake up an unconscious person in the field (only in medical facilitys or vehicles which are mostly too far away in our missions). If the patient isn't bleeding anymore, he has a normal pulse and blood presure and no pain it would be great if he would wake up automatically after some seconds (or by tapping him on his shoulder). Is there any possibility to configure that?

I know that there is a possibility to heal the soldier completely with a "personal aid kit". But you can always use this personal aid kid even if the soldier is bleeding, has pain, no blood presure or pulse and that sucks because this destroys the complete medical system when you could heal a soldier with only one item :-) Please correct me if I'm wrong.

So it would be great if we would have an option to wake up a healthy patient in the field. As example: A personal aid kit should only be usable when the soldier is completely fit (not bleeding, normal pulse and blood presure etc.) or the patient should wake up automatically if he is completely healthy or by tapping him on his shoulder.

Thank you very much :-)

**Comment by havena** [comment]
ho.. same question at same moment : https://github.com/acemod/ACE3/issues/1330

**Comment by esteldunedain** [comment]
>  But you can always use this personal aid kid even if the soldier is bleeding, has pain, no blood presure or pulse and that sucks because this destroys the complete medical system when you could heal a soldier with only one item :-) Please correct me if I'm wrong.

I think @glowbal talked recently about changing that behaviour, so the FAK only works with fully stabilized patients or something like that

**Comment by Knight1981** [comment]
Yeah that would be cool. At the moment this behaviour is really bad. 
Patient is completely fit but we can't wake him up or Patient is nearly dead and we can heal him just with one item like vanilla :D

Good to see that other player also wants to have this change :)

**Comment by sargken** [comment]
I would say make a timer system and depending on the injury make thr time dynamic.

**Comment by Macusercom** [comment]
I totally agree. Sometimes we play with only 8 people and for every unconsious person we have to fall back to the vehicle or a facility. Even if you want it to be true realism, then please add a module to allow waking up just like in CSE. This thing made us go back to CSE as the basic menu is too easy and the fact that you can't wake up makes it impossible for us to play.

**Comment by gienkov** [comment]
Lol, unconscious ppl wake up by themselves after few seconds, really. You need to stabilize their status, apply morphine, monitor heart rate and blood pressure and bandage all wounds of course, thats all. Never had issue that I needed to apply PAK to wake up somebody.

**Comment by Macusercom** [comment]
We tested it a dozen times and it didn't work for us. We did the same things as with CSE (bandage, then blood, epinephrine, CPR and maybe morphine when needed). The patient never wakes up.

**Comment by Knight1981** [comment]
@gienkov: Are you 100% sure that 100% of unconscious ppl wake up by themselves? I can't confirm this. In most cases the patient don't wake up by themselves even if he is absolutely fit. :-(

**Comment by Glowbal** [comment]
There is a difference between basic and advanced unconsciousness.

Basic:
- Unconscious until woken up by epinephrine.

Advanced:
- Unconscious until vitals are stable
- Massive bleeding is stopped/reduced
- Blood volume is high enough (>75% I think?)
- No massive amounts of pain (this is set to some incredible high level. If you reach it you are done anyway)
- Probably some other reasons.

**Comment by Macusercom** [comment]
@Glowbal We are using the advanced medic system and checked everything. Still no chance of waking up. How can we debug this? There must be something that isn't working or that a lot of us are doing wrong.

**Comment by GitHawk** [comment]
You can't revive players if they are in reviveState, even if they are otherwise fine:
```
hint format ["Revive State %1", player getVariable "ace_medical_inReviveState"];
```
The variable is only reset by FirstAidKit or manually by debug console.

**Comment by Dobakjang** [comment]
I can confirm that people can't revived even he is 100% in good vital condition, should i stop his heart rate and do cpr to wake him up? if that'so that's too much to revive one person

**Comment by GitHawk** [comment]
@Dobakjang You can't revive him without a FirstAidKit.

**Comment by ShuraDE** [comment]
GitHawk is right.

I cant find anything where the unit vitals or something else would be checked after the reviveState is set true.

Only ReviveTimer and ReviveLives >0 are checked
Often the settings are 9999 on timer, and -1 to Lives, so there never triggerd.

Is this really intended ?

I believe here https://github.com/acemod/ACE3/blob/master/addons/medical/functions/fnc_setDead.sqf#L65 should be a add check  for the vitals in, to reset reviveState and step back to setUnconscious, if there are fine.

**Comment by GitHawk** [comment]
@ShuraDE @Glowbal pointed out in #1330, that the current design is intended.
>It's this way because if you weren't using revive they would be dead. So it gets treated as a stage where the unit health state is that horrible the only way to do so would be by a casevac and treatment at a medical facility.

But I agree that "being in reviveState" should be easier identifiable (at least for medical staff).

**Comment by Macusercom** [comment]
Any updates on this? I still don't think that only being able to revive with a PAK makes the advanced system useful. Basically you'll never need anything else when the unit is unconsious.

**Comment by Glowbal** [comment]
If you are not able to wake a unit up in the field despite everything stable, it means the unit is in revive state. I've adjusted it so that vitals will always be very low (around 0) while a unit is in the revive state, despite all efforts.

Note that the revive state is something that is on top of the medical implementation - it only exist to give users the ability to extend the alive time of a unit.

Also, units that are in a revive state are in the unconscious state. They are not considered dead, but should have been where it not for the revive state. This is why we went with an implementation that requires more advanced treatment to wake up a patient - not just stablizing. 

I will look into more options to polish the revive and personal aid kit features for advanced medical.
